### PR TITLE
Convert short hex (#fff) to color

### PR DESCRIPTION
**Pull request by kana-sama**
`Color.Convert.hexToColor` now can convert `#fa0` to `( 255, 170, 0 )`

### CODE REVIEW

File: src/Color/Convert.elm [hunk 1]
```diff
@@ -133,22 +132,53 @@ Converts a string to `Maybe` of color.
...
+        extend token =
+            case String.toList token of
+                [ token ] ->
+                    String.fromList [ token, token ]
+
+                _ ->
+                    token
+
+        {- Converts "<pattern>" to "(<pattern><pattern><pattern>)" -}
+        buildColorPattern : String -> String
```
Comment (eskimoblood) [inline src/Color/Convert.elm]: Wouldn't it be simpler to just write the regex by hand?

File: src/Color/Convert.elm [hunk 2]
```diff
@@ -133,22 +132,53 @@ Converts a string to `Maybe` of color.
...
+                ++ buildColorPattern "([a-f\\d]{2})"
+                ++ "|"
+                ++ buildColorPattern "([a-f\\d])"
+                ++ ")$"
+    in
+        String.toLower
+            >> Regex.find (Regex.AtMost 1) (Regex.regex pattern)
+            >> List.map .submatches
+            >> List.head
+            >> Maybe.andThen List.tail
```
Comment (eskimoblood) [inline src/Color/Convert.elm]: Can you please explain what happens here cause I cant follow the code.

**Review by eskimoblood — COMMENTED** [review]
Thanks for the pull request. As I like to understand the code before merging it back please answer my two question in the code comments.

File: src/Color/Convert.elm (hunk 1 above)
Comment (kana-sama) [reply src/Color/Convert.elm]: What about this?
```elm
pattern =
    ""
        ++ "^"
        ++ "#?"
        ++ "("
        ++ "(([a-f\\d]{2})([a-f\\d]{2})([a-f\\d]{2}))"
        ++ "|"
        ++ "(([a-f\\d])([a-f\\d])([a-f\\d]))"
        ++ ")"
        ++ "$"
```

File: src/Color/Convert.elm [hunk 3]
```diff
@@ -133,22 +132,44 @@ Converts a string to `Maybe` of color.
...
+                ++ "|"
+                ++ "(?:([a-f\\d])([a-f\\d])([a-f\\d]))"
+                ++ ")"
+                ++ "$"
+    in
+        String.toLower
+            >> Regex.find (Regex.AtMost 1) (Regex.regex pattern)
+            >> List.head
+            >> Maybe.map .submatches
+            >> Maybe.map (List.filterMap identity)
```
Comment (kana-sama) [inline src/Color/Convert.elm:161]: There are six groups, three of them are Nothing (3 tokens for short or 3 tokens for full form).

File: src/Color/Convert.elm (hunk 2 above)
Comment (kana-sama) [reply src/Color/Convert.elm]: refactored

**Comment by eskimoblood** [comment]
Thanks for the changes, I will merge it after work

**Comment by kana-sama** [comment]
I'm not very good in English, so I can't choose a good name for function, which complements token (`f -> ff`, `ff -> ff`)

**Pull request merged**